### DISCUSSION TITLE
[Ata da reuniao] 17 de maio de 2021, 21h

### DISCUSSION THREAD

**Discussion by jenrodrigues** [andrebaltieri/AgendaEventos#51] [General]
### **Quem participou:**
Andre Baltieri
Jenifer Rodrigues
Gabriel Nalin
Luan Freitas
Joao Colussi
Brewerton Santos
Leonardo Pedrosa

### **Topicos debatidos:**
1) Balta revisou o script de banco de dados criado pelo time dev.

### **Proximos passos:**
1) Balta pediu que as seguintes mudancas sejam efetuadas no script:
- Tabela Event, campo Description: mudar o tipo do campo para TEXT
- Tabela Event, campo Url: mudar o tamanho do campo para 1024
- Tabela Event, campo Url: mudar o nome do campo para UrlSegment
- Tabela Event, campo Banner: mudar o tamanho do campo para 2000
- Tabela Event, campo Banner: mudar o nome do campo para BannerUrl
- Tabela Event, campo Active: configurar default value do campo para False
- Tabela AttendeeEvents: corrigir o nome da chave estrangeira (deveria ser EventId, mas esta EventI)
- Avaliar as tabelas que precisam de indices e cria-los.

2) Atualizar a classe Event.cs com as mudancas de nome de campos sugeridas no item 1.

3) Fazer/revisar o curso “Fundamentos do Git e Azure DevOps” (https://balta.io/cursos/fundamentos-git-azure-devops).
 
4) Ler/revisar o artigo “WSL - Windows Subsystem for Linux” (https://balta.io/blog/wsl).

5) Ler/revisar o artigo “SQL Server Docker” (https://balta.io/blog/sql-server-docker). Pra quem e aluno Premium, a criacao de um container Docker utilizando uma imagem do SQL Server tambem e demonstrada no curso “Criando APIs Data Driven com ASP.NET Core 3 e EF Core 3” (https://balta.io/cursos/criando-apis-data-driven-com-aspnet-core-3-e-ef-core-3). 

### **Proxima reuniao em 24 de maio as 21h. Topicos a serem debatidos:**
- Apresentar ao Balta as mudancas realizadas no script e nas classes.
- Levantar possiveis duvidas sobre as recomendacoes de estudos (itens 3, 4 e 5).

**Comment by GabrielNalin** [comment]
Alterações feitas porém ainda persistem algumas dúvida:
Na Tabela Event, a quantidade de 2000 caracteres do campo "Description" é mesmo suficiente?
Na Tabela Event, o campo Active configurado default para False é o correto mesmo? 
Após análise da equipe, farei o ítem 2 da ata "Atualizar a classe Event.cs com as mudancas de nome de campos sugeridas no item 1".

**Reply by jenrodrigues** [reply]
No meu entendimento, o campo TEXT nao precisa da definicao de caracteres. Foi por isso que o Balta pediu esse tipo. Pra garantir que o campo aceite um conteudo texto o mais longo possivel, caso precisemos.
Balta pediu o default value False porque no entendimento dele todo evento deve ser criado de forma "inativa" e ativado depois, para garantir que determinados eventos nao aparecam para o usuario final acidentalmente (por exemplo, com dados ainda sendo editados).

**Reply by jenrodrigues** [reply]
Nao estou certa de que a criacao do default value pro campo ACTIVE esta correta. Um exemplo: https://stackoverflow.com/questions/43996465/bit-type-default-to-0-instead-of-null

**Reply by renebentes** [reply]
Opa. Pegando o bonde andando aqui. Mas estou de acordo com a @jenrodrigues, o campo deveria ser [Active] BIT IS NOT NULL DEFAULT 0. Quanto aos índices creio que as tabelas de Eventos, Categorias e Usuários são boas candidatas, principalmente a de eventos.

**Reply by GabrielNalin** [reply]
Boa tarde,
Como o Balta pediu active "FALSE", vamos deixar FALSE mesmo e conversar com ele na proxima reunião? No caso de mudança, fazemos na hora.
Retirei a quantidade de caracteres do campo TEXT.

**Reply by jenrodrigues** [reply]
@renebentes eu acho que as colunas candidatas a indice nas tabelas Category e User sao primary key e/ou unique identifier. O Balta fala no curso de Fundamentos SQL Server que nao precisa criar indices pra esse tipo de coluna porque elas ja seriam originalmente indices. Voce sabe alguma coisa sobre isso?

**Reply by JoaoNascSilva** [reply]
Bom dia pessoal, geralmente índices são para melhoria de consultas em determinadas colunas (otimização). 
Então creio que poderíamos estar analisando e criando para aquelas tabelas e colunas que podem vir a ter mais acessos(consultas).

**Comment by GabrielNalin** [comment]
Então eu q entendi errado. Vou retirar a quantidade de caracteres do campo
texto.
A dúvida sobre o default value false está ok resolvida pra mim.



Em qua, 19 de mai de 2021 13:36, Jenifer Rodrigues ***@***.***>
escreveu:

**Comment by GabrielNalin** [comment]
Item 2 da ata realizado (2.0Atualizar a classe Event.cs com as mudancas de nome de campos sugeridas no item 1.)
